### DISCUSSION TITLE
modularized ::  Detection From One Camera on ALL

### DISCUSSION THREAD

**Discussion by gniknalu** [roflcoopter/viseron#328] [Q&A] [answered]
Sad to see this return even in the modularized version. With a 'detection' at 06:53 everything was fine. (The word detection was in quotes because it was really the camera switching from night to day.) Then at 08:38 the attached image was 'detected'. (Not sure what was detected.) But notice it shows 4 detections from the other camera and none from this one
![2022-02-15-083808](https://user-images.githubusercontent.com/3652809/154121942-634061f1-5c7b-4812-bc50-3410e2845d1e.jpg)

I reloaded it and it seems to have fixed itself but I can't verify as I can't see the detections live.

**Comment by roflcoopter** [comment]
This shouldnt be possible.... i wrapped all calls to subprocesses in a lock to prevent this.
I cant reproduce it either anymore. 

Its a bug in OpenCV that they dont seem to fix sadly 
https://github.com/opencv/opencv/issues/19643

When you say reload, you just restarted the container?

**Reply by gniknalu** [reply]
> When you say reload, you just restarted the container?

Yes - restart the container. Happened again this morning.

![2022-02-16-065304](https://user-images.githubusercontent.com/3652809/154282118-8f0a11d0-15b3-4c1f-b167-99c153888587.jpg)

**Reply by gniknalu** [reply]
> This shouldnt be possible.... i wrapped all calls to subprocesses in a lock to prevent this. I cant reproduce it either anymore.
> 
> Its a bug in OpenCV that they dont seem to fix sadly [opencv/opencv#19643](https://github.com/opencv/opencv/issues/19643)
> 
> When you say reload, you just restarted the container?

I noticed that OpenCV issue is closed?  . . .

**Reply by roflcoopter** [reply]
Yeah its closed but unfortunately not fixed.
I even commented there but received no reply

**Comment by roflcoopter** [comment, marked as answer]
I just noticed this issue arising again.
Tbh i was SO SURE i had fixed it that i refused to believe you had this problem 😂

I decided to ditch OpenCV altogether and move to native darknet instead.
I am currently running some tests and it seems to work well, slightly better performance even.
And most notably, NO duplicate detections.

Ill come back here and let you know when its pushed

**Reply by gniknalu** [reply]
> I decided to ditch OpenCV altogether and move to native darknet instead. I am currently running some tests and it seems to work well, slightly better performance even. And most notably, NO duplicate detections.
> 
> Ill come back here and let you know when its pushed

**Thank you!**

**Reply by roflcoopter** [reply]
These changes are now in the latest version of `roflcoopter/amd64-cuda-viseron:modularize`, when you get the time, could you try and see how it works for you?

**Reply by gniknalu** [reply]
> These changes are now in the latest version of `roflcoopter/amd64-cuda-viseron:modularize`, when you get the time, could you try and see how it works for you?

I've been using it for the past 5 hours and it's awesome! No duplicate detections. Accurate detections. This is beautiful!
